pvanalytics/quality/data_shifts.py

```diff
@@ -0,0 +1,335 @@
+        'cosine', or 'linear'. See the following documentation for further
+        information: https://centre-borelli.github.io/ruptures-docs/user-guide/
+        Default set to "rbf".
+    penalty: int.
+        Penalty value passed to the ruptures changepoint detection method.
+        Default set to 40.
+
+    Returns
+    -------
+    list
+        The returned list contains the pandas timestamps where data shifts were
+        detected.
+    """
+    # Run data checks on cleaned data to make sure that the data can be run
+    # successfully through the routine
+    _run_data_checks(time_series, use_default_models, method, cost, penalty)
+    # Run the filtering sequence, if marked as True
+    if filtering:
+        time_series = _erroneous_filter(time_series)
+    # Check if the time series is more than 2 years long. If so, remove
+    # seasonality. If not, run analysis on the normalized time series
+    if (time_series.index.max() - time_series.index.min()).days <= 730:
+        warnings.warn("The passed time series is less than 2 years in length, "
+                      "and cannot be corrected for seasonality. Runnning data "
+                      "shift detection on the min-max normalized time series "
+                      "(NO seasonality correction).")
+        time_series_processed = _preprocess_data(time_series,
+                                                 remove_seasonality=False)
+        seasonality_rmv = False
+    else:
+        # Perform pre-processing on the time series, to get the
+        # seasonality-removed time series.
+        time_series_processed = _preprocess_data(time_series,
+                                                 remove_seasonality=True)
+        seasonality_rmv = True
+    points = np.array(time_series_processed.dropna())
+    # If seasonality has been removed and default model is used, run
+    # BottomUp method
+    if (seasonality_rmv) & (use_default_models):
+        algo = rpt.BottomUp(model='rbf').fit(points)
+        result = algo.predict(pen=40)
+    # If there is no seasonality but default model is used, run
+    # Window-based method
+    elif (not seasonality_rmv) & (use_default_models):
+        algo = rpt.Window(model='rbf',
+                          width=50).fit(points)
+        result = algo.predict(pen=30)
+    # Otherwise run changepoint detection with the passed parameters
+    else:
+        algo = method(model=cost).fit(points)
+        result = algo.predict(pen=penalty)
+    # Remove the last index of the time series, if present
+    if len(points) in result:
+        result.remove(len(points))
+    # Return a list of dates where changepoints are detected
+    time_series_processed.index.name = "datetime"
+    time_series_processed = time_series_processed.reset_index()
+    return list(time_series_processed.loc[result]['datetime'])
+
+
+def filter_data_shifts(time_series, filtering=True, use_default_models=True,
```

Comment:
If `detect_data_shifts` returns a Series of boolean, then this function can return the (start, end) pairs defining intervals of change points, or, just the indices of the change points. I see value in that.  I agree that `filter_` implies that it would return a subset of the input Series.